solr/core/src/java/org/apache/solr/core/TracerConfigurator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.core;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Map;
+
+import io.opentracing.Tracer;
+import io.opentracing.noop.NoopTracerFactory;
+import org.apache.solr.common.cloud.ClusterPropertiesListener;
+import org.apache.solr.common.cloud.ZkStateReader;
+import org.apache.solr.util.plugin.NamedListInitializedPlugin;
+import org.apache.solr.util.tracing.GlobalTracer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class TracerConfigurator implements NamedListInitializedPlugin {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+  public abstract Tracer getTracer();
+
+  public static void loadTracer(SolrResourceLoader loader, PluginInfo info, ZkStateReader stateReader) {
+    if (info == null) {
+      GlobalTracer.setup(NoopTracerFactory.create());
+      GlobalTracer.setSamplePercentage(0.0);
+    } else {
+      TracerConfigurator configurator = loader
+          .newInstance(info.className, TracerConfigurator.class);
+      configurator.init(info.initArgs);
+
+      GlobalTracer.setup(configurator.getTracer());
+
+      stateReader.registerClusterPropertiesListener(properties -> {
+        if (properties.containsKey(ZkStateReader.SAMPLE_PERCENTAGE)) {
+          try {
+            double sampleRate = Double.parseDouble(properties.get(ZkStateReader.SAMPLE_PERCENTAGE).toString());
+            GlobalTracer.setSamplePercentage(sampleRate);
+          } catch (NumberFormatException e) {
+            log.error("Unable to set sample rate", e);
+          }
+        } else {
+          GlobalTracer.setSamplePercentage(1);
```

Comment:
Tracing at 100% could generate a lot of trace data. The default sample rate for many tracers like Zipkin is 0.1%. Do we want to use a lower default or do we really want to trace everything in the event that no explicit rate is defined?

Reply:
This means 1% instead of 100%, but 0.1% seems more reasonable.